modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts

```diff
@@ -1945,6 +1912,90 @@ export class NotebookCtrl {
         };
     }
 
+    scanActions: QueryActions<Paragraph & {type: 'scan'}> = [
+        {
+            text: 'Scan',
+            click: (p) => this.$scope.scan(p),
+            available: (p) => this.$scope.scanAvailable(p)
+        },
+        {
+            text: 'Scan on selected node',
+            click: (p) => this.$scope.scan(p, true),
+            available: (p) => this.$scope.scanAvailable(p)
+        },
+        {text: 'Rename', click: (p) => this.renameParagraph(p), available: () => true},
+        {text: 'Remove', click: (p) => this.removeParagraph(p), available: () => true}
+    ]
+
+    queryActions: QueryActions<Paragraph & {type: 'query'}> = [
+        {
+            text: 'Execute',
+            click: (p) => this.$scope.execute(p),
+            available: (p) => this.$scope.queryAvailable(p)
+        },
+        {
+            text: 'Execute on selected node',
+            click: (p) => this.$scope.execute(p, true),
+            available: (p) => this.$scope.queryAvailable(p)
+        },
+        {
+            text: 'Explain',
+            click: (p) => this.$scope.explain(p),
+            available: (p) => this.$scope.queryAvailable(p)
+        },
+        {text: 'Rename', click: (p) => this.renameParagraph(p), available: () => true},
+        {text: 'Remove', click: (p) => this.removeParagraph(p), available: () => true}
+    ]
+
+    async renameParagraph(paragraph: Paragraph) {
+        const newName = await this.IgniteInput.input('Rename Query', 'New query name:', paragraph.name);
+
+        if (!newName)
```

Comment:
I think this check is not necessary, because IgniteInput should have 'required' validation